### PR TITLE
Adding support for subcategories

### PR DESCRIPTION
**Pull request by riknoll**
In addition to the common "More" category.

### CODE REVIEW

**Comment by pelikhan** [comment]
Why do we need those?

**Comment by pelikhan** [comment]
We should guarantee a constant ordering in sub-categories.

**Comment by pelikhan** [comment]
Feels like we should simply use the namespace information.

**Comment by riknoll** [comment]
@pelikhan I agree on having a constant ordering, but using the namespace information makes the signature pretty ugly:

```typescript
light.builtin.showAnimationFrame(light.animation.sparkleAnimation());
```

Kids might find that somewhat confusing considering `light.builtin` is an object and `light.animation` is a namespace.

**Comment by pelikhan** [comment]
```light.animation.sparkle()``` does not sound too bad. we can support that later.

File: pxtlib/emitter/emitter.ts [hunk 1]
```diff
@@ -301,6 +301,7 @@ namespace ts.pxtc {
         parts?: string;
         trackArgs?: number[];
         advanced?: boolean;
+        subcategory?: string;
```
Comment (pelikhan) [inline pxtlib/emitter/emitter.ts]: field for this interface typically use a camelized notation.

Comment (pelikhan) [inline pxtlib/emitter/emitter.ts]: otherwise LGTM

Comment (riknoll) [inline pxtlib/emitter/emitter.ts]: subcategory is one word

**Comment by riknoll** [comment]
I'm adding support for specifying subcategory weights, so hold off on merging

File: pxtlib/emitter/emitter.ts (hunk 1 above)
Comment (pelikhan) [inline pxtlib/emitter/emitter.ts]: goodToGoThen!

**Comment by riknoll** [comment]
@pelikhan how about this:

1. Subcategories specified by the parent package always come first and are weighted by the package
2. Subcategories from other packages come next and are sorted alphabetically
3. "More" always goes last

**Comment by pelikhan** [comment]
Could we simply get rid of the built in ones? We just need to have a sufficient weight. This is how all the other categories work...

**Comment by riknoll** [comment]
You mean sort them all alphabetically? "More" should remain as is because it should be consistent across parent categories.

**Pull request closed without merging**